evalml/data_checks/invalid_target_data_check.py

```diff
@@ -44,10 +52,18 @@ def __init__(self, problem_type, objective, n_unique=100):
         self.n_unique = n_unique
 
     def validate(self, X, y):
-        """Check if the target data contains missing or invalid values.
+        """Check if the target data is considered invalid.
+
+        Target data is considered invalid if:
+            - Target is None.
+            - Target has NaN or None values.
+            - Target is of an unsupported Woodwork logical type.
+            - Target and features have different lengths or indices.
+            - Target does not have enough instances of a class in a classification problem.
+            - Target does not contain numeric data for regression problems.
 
         Args:
-            X (pd.DataFrame, np.ndarray): Features. Ignored.
+            X (pd.DataFrame, np.ndarray): Features. If not None, will be used to check that the target and features have the same dimensions and indices.
```

Comment:
Maybe it's redundant, but perhaps some mention of this might also be in the main docstring?  This is a weird data check as it deals mostly with the target but its first arg is X, which is ignored, lol.  Sometimes.